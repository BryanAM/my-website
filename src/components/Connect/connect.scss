@import '../../resources/styles/colors.scss';
@import '../../resources/styles/text.scss';
@import '../../resources/styles/screen.scss';
 
.connect-employed {
  color: $white;
  font-size: $small;
  color: $standard-text;
  margin-bottom: 50px;
  @media screen and (min-width: $break-large) {
    font-size: $x-large;
  }
}

.connect-hi {
  display: inline-block;
  border: solid 1px $accent-pink;
  border-radius: 0.5rem;
  padding: 1rem 2rem;
  margin: 0.25rem;
  min-width: 10rem;
  height: auto;
  align-self: center;
  text-decoration: none;
  background: $nav-blue;
  color: #ffffff;
  font-family: sans-serif;
  font-size: $medium;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  transition: all .5s ease-in-out;
  &:hover,
  &focus {
    background: rgba(230, 103, 167, 0.3);
  }
  
}

#connect-section {
  display: flex;
  flex-direction: column;
  min-height: 70vh;
}

.connect-title {
  color: $white;
  font-size: $large;
  margin-bottom: 0;
  font-weight: $x-bold;
  @media screen and (min-width: $break-large) {
    font-size: $xx-large;
  }
}