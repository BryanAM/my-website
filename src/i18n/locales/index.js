import en from './en.json';
import jp from './jp.json';
import ko from './ko.json';

export {
  en,
  jp,
  ko
}