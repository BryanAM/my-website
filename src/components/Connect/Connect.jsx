import React from 'react';
import { useTranslation } from 'react-i18next';
import './connect.scss';
const Connect = () => {
  const [t] =useTranslation();
  const handleOnClick = () => {
    window.location = "mailto:bryaument@gmail.com?subject=I saw your website! I'm saying hi!";
  }
  return(
    <section id='connect-section'>
      <h2 className='section-header'>{t('connect.header')}</h2>
      <h2 className='connect-title'>{t('connect.me')}</h2>
      <p className='connect-employed'>{t('connect.employed')}</p>
      <button className='connect-hi' onClick={handleOnClick}>{t('connect.hi')}</button>
    </section>
  );
};
export default Connect;