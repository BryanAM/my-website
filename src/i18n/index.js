import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { en, jp, kr } from './locales/index';

const options = {
  interpolation: {
    escapeValue: false,
  },
  caches: ['localStorage', 'cookie'],
  fallbackLng: 'en',
  debug: true,

  resources: {
    en: {
      common: en.en,
    },
    
    // default japanese ISO code = ja
    ja: {
      common: jp.jp
    },

    kr: {
      common: kr.kr
    }
  },

  ns: ['common'],
  defaultNS: 'common',

  react: {
    wait: false,
    bindI18n: 'languageChanged loaded',
    bindStore: 'added removed',
    nsMode: 'default',
  },
};

i18n
  .use(LanguageDetector)
  .init(options)

export default i18n;