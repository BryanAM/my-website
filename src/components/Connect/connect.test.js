import React from 'react'
import { shallow, mount } from 'enzyme';
import i18n from '../../i18n/index.js';
import Connect from './Connect.jsx';

describe('Connect', () => {
  it('should render on load correctly', () => {
    const component = shallow(<Connect />);
    expect(component).toMatchSnapshot();
  });
}); 
