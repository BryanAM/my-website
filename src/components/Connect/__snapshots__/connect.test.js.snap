// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect should render on load correctly 1`] = `
<section
  className="section"
  id="connect-section"
>
  <h2
    className="section-header"
  >
    connect.header
  </h2>
  <h2
    className="connect-title"
  >
    connect.me
  </h2>
  <p
    className="connect-employed"
  >
    connect.employed
  </p>
  <button
    className="connect-hi"
    onClick={[Function]}
  >
    connect.hi
  </button>
</section>
`;
